textures/src/main/java/com/dumbdogdiner/stickyapi/util/textures/TextureHelper.java

```diff
@@ -0,0 +1,259 @@
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new RuntimeException("An unknown error occurred while accessing the builtin resource 'textures.json'.", e);
+            } catch (ClassCastException e) {
+                throw new RuntimeException("The integrated textures.json resource was invalid. Please check the format at compile-time, ensure that the resource is generated correctly. If you are a server owner, contact the developers of StickyAPI", e);
+            }
+
+        } catch (RuntimeException e) {
+            Bukkit.getLogger().severe(e.getMessage());
+            throw e;
+        }
+    }
+
+    /**
+     * Gets the {@link Set} of categories of textures in the Heads file
+     *
+     * @return A {@link Set} of all the categories of texture
+     */
+    public static @NotNull Set<String> getCategories() {
+        return TextureMap.keySet();
+    }
+
+    /**
+     * Get a {@link Set} of textures
+     *
+     * @param cat The {@link #getCategories() category}of textures to get the {@link Map}
+     * @return A {@link Map} of texture names to texture strings for the given category
+     * @see TextureHelper#getCategories()
+     */
+    public static Map<String, String> getTextureMapCategory(@NotNull String cat) {
+        return TextureMap.get(cat.toUpperCase());
+    }
+
+
+    /**
+     * Get a list of all the textures for a given Category
+     *
+     * @param cat a Category of textures
+     * @return a {@link List} of the textures in a given Category
+     * @see #getCategories()
+     */
+    public static @NotNull List<String> getTextures(@NotNull String cat) {
+        return Collections.unmodifiableList(new ArrayList<>(getTextureMapCategory(cat).keySet()));
+    }
+
+    /**
+     * Gets a texture string given a category and texture name
+     *
+     * @param cat  The texture category
+     * @param name The texture name
+     * @return The texture string matching
+     * @throws NoSuchElementException if the specified texture is not found
+     */
+    public static String getTexture(@NotNull String cat, @NotNull String name) throws NoSuchElementException {
+        return TextureMap.get(cat.toUpperCase()).get(name.toUpperCase());
+    }
+
+    /**
+     * Gets a texture string given a qualified name
+     *
+     * @param qualifiedName A given texture as a qualified name
+     * @return the texture string for the given texture
+     * @throws NoSuchElementException if the specified texture is not found
+     * @see #toQualifiedName(String, String)
+     */
+    public static @Nullable String getTexture(@NotNull String qualifiedName) throws NoSuchElementException {
+        String[] splits = qualifiedName.split("\\.");
+        if (splits.length != 2 && splits.length != 1)
+            throw new RuntimeException("Invalid qualified name: " + qualifiedName);
+        if (splits[0].equals("*")) {
+            @Nullable String texture = null;
```

Comment:
Does IntelliJ complain about this if you don't use it? I mostly use the null safety annotations for Kotlin's sake